packages/labs/analyzer/src/lib/javascript/jsdoc.ts

```diff
@@ -51,6 +52,16 @@ const parseNameDescRE = /^(?<name>^\S+)(?:\s?-\s+)?(?<description>[\s\S]*)$/m;
 const parseNameDashDescRE =
   /^(?<name>^\S*)?(?:\s+-\s+(?<description>[\s\S]*))?$/m;
 
+// Regex for parsing name and description from JSDoc comments
+const parseNameDefaultDescRE =
+  /^\[(?<name>[^[\s=]*)=(?<defaultValue>[^\]]*)\](?:\s+-\s+(?<description>[\s\S]*))?/m;
```

Comment:
do you also want to account for optional css props without defaults? for consistency with `@param` and other built-ins?

```ts
/**
 * @cssProp [--some-optional-css-prop]
 * @cssProp --some-required-css-prop
 * @cssProp [--some-optional-css-prop-with-value=123]
 */
```

the square brackets usually mean "optional". you can only have default values for optional things, so thats why you also only ever see default values in such positions

may be worth keeping the supported syntax aligned with the regular jsdoc syntax

also the `\s\S` is basically behaving as `.+` here right? the description is already optional so you could dumb it down to `(?<description>.+)?`

the dash also looks required here but your post suggests it isn't. hows that working?

Reply:
I think `\s\S` is a typo
Dash shouldn't be required I'll take a look
I'm not sure what the meaning of "optional but without default" is in css world, but maybe `@property` makes that a runtime thing so yeah i'll add that

Thanks for the review!

Reply:
yeah in the css world it doesn't make much sense to have an optional vs required property (they're all optional...).

however i imagine a fair amount of people will assume you can do `cssProp --foo` rather than `cssProp [--foo]`.

`@param` and friends already define this behaviour well enough so we may as well follow how they work, even if it doesn't apply as much in css-world

Reply:
ah and also you probably want to use `+` rather than `*` on the name and default value since you can't have an empty one (`[]`)

so `(?<name>[^[\s-]+)=(?<defaultValue>[^\]]+)`

Reply:
PTAL